### PR TITLE
chore: add docker buildx xtask

### PR DESCRIPTION
**Pull request by b5**
much easier to do multi-arch builds with buildx, at the cost of taking eons & needing every resource the computer has. This is mainly here to document the technique for later inclusion in CI, and for my own quality of life

### CODE REVIEW

**Comment by Arqu** [comment]
> much easier to do multi-arch builds with buildx, at the cost of taking eons & needing every resource the computer has. This is mainly here to document the technique for later inclusion in CI, and for my own quality of life

Imma go cry in my own sad CI corner :D

File: xtask/src/main.rs [hunk 1]
```diff
@@ -196,6 +213,65 @@ fn build_docker(all: bool, build_images: Vec<String>, progress: String) -> Resul
...
+                ".",
+            ])
+            .status()?;
+
+        if !status.success() {
+            Err(anyhow::anyhow!("docker buildx failed"))?;
+        }
+    }
+
+    // let status = Command::new("docker")
```
Comment (Arqu) [inline xtask/src/main.rs:259]: Can we drop dead code?

**Review by Arqu — APPROVED** [review]
LGTM, `buildx` is definitely the way to go.

File: xtask/src/main.rs (hunk 1 above)
Comment (b5) [reply xtask/src/main.rs:259]: normally yes, but in this case I'd like to have it in commit history, and build upon it in a future iteration. If buildx is the way to go, I'll need to put more time into this command soon

**Comment by Arqu** [comment]
`buildx` is for building docker images for different platforms from your host. Code incidentally also compiles for the dockerized platform, but you kinda get the same thing.

**Pull request merged**